### PR TITLE
Add gRPC compiler plugin

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/814

## Automation tests
 - Unit tests 
   > Yes

## Test environment
- macOS
- Java 11
- SLAlpha 3 and 4

### CODE REVIEW

File: grpc-ballerina/grpc_errors.bal [hunk 1]
```diff
@@ -81,7 +81,7 @@ public type DataMismatchError distinct Error;
 public type ClientAuthError distinct Error;
 
 # Represents an error when reaching the end of the client stream.
-public type EOS distinct Error;
+public type EosError distinct Error;
```
Comment (daneshk) [inline grpc-ballerina/grpc_errors.bal:84]: we can remove this error type now. we are not using anywhere

Comment (BuddhiWathsala) [reply grpc-ballerina/grpc_errors.bal]: Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/154/commits/0203e25d8289ee970a44df37dd233c69f099dac1

File: grpc-compiler-plugin-tests/src/test/resources/testng.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="BallerinaGraphqlCompilerPluginTests">
+    <test name="UnitTests">
+        <classes>
+            <class name="org.ballerinalang.net.grpc.plugin.CompilerPluginTest"/>
+        </classes>
+    </test>
+</suite>
```
Comment (daneshk) [inline grpc-compiler-plugin-tests/src/test/resources/testng.xml]:
```suggestion
</suite>

```

Comment (BuddhiWathsala) [reply grpc-compiler-plugin-tests/src/test/resources/testng.xml]: Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/154/commits/ab095673558f46092afb3a57bc6855487df28315

File: grpc-compiler-plugin/src/main/java/org/ballerinalang/net/grpc/plugin/GrpcServiceValidator.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                    validateFunctionSignature(functionDefinitionNode, syntaxNodeAnalysisContext, serviceName);
+
+                });
+            }
+
+        }
+
+    }
+
+    public boolean isBallerinaGrpcService(ServiceDeclarationSymbol serviceDeclarationSymbol) {
```
Comment (daneshk) [inline grpc-compiler-plugin/src/main/java/org/ballerinalang/net/grpc/plugin/GrpcServiceValidator.java:91]: Shall we make these local functions `private`

Comment (BuddhiWathsala) [reply grpc-compiler-plugin/src/main/java/org/ballerinalang/net/grpc/plugin/GrpcServiceValidator.java]: Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/154/commits/b419385aa5fdaaa4d2bfafdf82ef1093e5f0b9c3

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**